libs/clients/fiskistofa/src/lib/fiskistofaClient.service.ts

```diff
@@ -0,0 +1,143 @@
+import { Injectable } from '@nestjs/common'
+import {
+  GetShipsInput,
+  GetQuotaTypesForCalendarYearInput,
+  GetQuotaTypesForTimePeriodInput,
+  UpdateShipStatusForCalendarYearInput,
+  GetShipStatusForCalendarYearInput,
+  GetShipStatusForTimePeriodInput,
+  UpdateShipStatusForTimePeriodInput,
+  UpdateShipQuotaStatusForTimePeriodInput,
+  GetSingleShipInput,
+} from '@island.is/api/domains/fiskistofa'
+import {
+  V1SkipSkipnumerGetRequest,
+  V1StadaskipsKvotategundirAlmanaksarArGetRequest,
+  V1StadaskipsKvotategundirFiskveidiarFiskveidiarGetRequest,
+  V1StadaskipsSkipnumerAlmanaksarArDeilistofnarBreyttPostRequest,
+  V1StadaskipsSkipnumerAlmanaksarArDeilistofnarGetRequest,
+  V1StadaskipsSkipnumerFiskveidiarFiskveidiarBreyttPostRequest,
+  V1StadaskipsSkipnumerFiskveidiarFiskveidiarGetRequest,
+  V1StadaskipsSkipnumerFiskveidiarFiskveidiarKvotiBreyttPostRequest,
+} from '../../gen/fetch'
+import { FetchError } from '@island.is/clients/middlewares'
+import { FiskistofaApi } from './api'
+
+@Injectable()
+export class FiskistofaClientService {
+  constructor(private api: FiskistofaApi) {}
+
+  async wrapper(callback: (api: FiskistofaApi) => any) {
+    try {
+      this.api.initialize()
+      return callback(this.api)
+    } catch (error) {
+      if (error instanceof FetchError) {
+        if (error.status === 401) {
+          this.api.initialize(true)
```

Comment:
It'd be good to see some comments here, why this is needed.

Reply:
I added a comment
`If we are unauthorized, then we want to re-initialize the api by getting a new access token and then calling the callback again`